cluster/addons/rbac/cloud-controller-manager-roles.yaml

```diff
@@ -0,0 +1,113 @@
+apiVersion: v1
+items:
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: ClusterRole
+  metadata:
+    name: cloud-controller-manager
+  rules:
+  - apiGroups:
```

Comment:
A lot of these permissions are for when we don't use service accounts per controller. The CCM itself needs a subset of these. If i'm not mistaken it just needs get/list/watch for resources (informers), SA creation, events and endpoints.

Reply:
in my opinion, the role should include the permissions you'd need to run the control loop in a pod

Reply:
@liggitt So you're saying the root service account should also have all the permissions of all the controllers?

Reply:
I agree with @jhorwit2... a controller manager should have permissions needed to launch its control loops. that typically means:
* service account setup permissions (read/write service accounts, read access to secrets)
* shared informer permissions (read access to relevant resources)

and that's about it

Reply:
Interestingly I found an orthogonal set of permissions needed to successfully start the CCM. I think rather than having a static list/yaml files in cluster/addons, we would be better of having a list/yaml files in a CP directory and the CP start script should package/copy the list/yaml files into cluster/addons. This also seems like it will more naturally work toward our goal state of allowing each CP to control things like what controllers they run and how many CMs are used to run those controller. It also means the CPs can sign in to having extra permissions as they get things like CM online. So for instance as most CPs don't yet have a CCM running they won't need to add these permissions yet. When they do add them they can better control the permissions given out.